antarest/study/service.py

```diff
@@ -1060,6 +1061,28 @@ def sync_studies_on_disk(
                 if self.storage_service.raw_study_service.update_name_and_version_from_raw_meta(existing_study):
                     self.repository.save(existing_study)
 
+    def delete_missing_studies(self) -> None:
+        """
+        Removes from the database any non-archived desktop studies whose folders
+        no longer exist on disk. Used to clean up orphaned study entries.
+        """
+        all_studies = self.repository.get_all_raw()
+        desktop_studies = (
+            study
+            for study in all_studies
+            if study.workspace != DEFAULT_WORKSPACE_NAME and isinstance(study, RawStudy) and not study.archived
+        )
+
+        def get_path(study: RawStudy) -> Path:
+            wp = self.config.get_workspace_path(workspace=study.workspace)
+            return wp / str(study.folder)
+
+        missing_studies = (study for study in desktop_studies if not is_study_folder(get_path(study)))
+
+        # delete orphan studies on database
+        for study in missing_studies:
+            self.repository.delete(study.id)
```

Comment:
I think you can pass all the studies at one according to the signature of the delete method. So we perform only one commit